### PR TITLE
Prevent detection of 8 and 9 as octal values

### PR DESCRIPTION
**Pull request by julienfalque**
Fixes https://github.com/adrienverge/yamllint/issues/271#issuecomment-688237384.

### CODE REVIEW

**Review by adrienverge — COMMENTED** [review]
Looks good, thanks!

File: yamllint/rules/octal_values.py [hunk 1]
```diff
@@ -83,6 +85,8 @@
 DEFAULT = {'forbid-implicit-octal': True,
            'forbid-explicit-octal': True}
```
[the remark this answers is not in the document]
Comment (julienfalque) [reply yamllint/rules/octal_values.py:87]: Done.

File: yamllint/rules/octal_values.py [hunk 2]
```diff
@@ -92,7 +96,7 @@ def check(conf, token, prev, next, nextnext, context):
         if isinstance(token, yaml.tokens.ScalarToken):
             if not token.style:
                 val = token.value
-                if val.isdigit() and len(val) > 1 and val[0] == '0':
+                if val.isdigit() and len(val) > 1 and val[0] == '0' and _is_octal_number(val):
```
[the remark this answers is not in the document]
Comment (julienfalque) [reply yamllint/rules/octal_values.py]: Done.

File: yamllint/rules/octal_values.py [hunk 3]
```diff
@@ -83,6 +85,8 @@
 DEFAULT = {'forbid-implicit-octal': True,
            'forbid-explicit-octal': True}
 
+def _is_octal_number(string):
+    return re.match(r'^[0-8]+$', string) is not None
```
[the remark this answers is not in the document]
Comment (julienfalque) [reply yamllint/rules/octal_values.py]: Woops :)

**Review by adrienverge — APPROVED** [review]
Thank you for contributing!

**Pull request merged**